clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -430,6 +430,9 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
       unsigned Minor = Version.getMinor().value_or(0);
       Builder.defineMacro("__SHADER_TARGET_MINOR", Twine(Minor));
     }
+    if (TI.getTriple().isSPIRV()) {
+      Builder.defineMacro("__spirv__");
```

Comment:
This seems like an odd place to define this macro when other targets tend to define them in clang/lib/Basic/Targets/`
for an example see:
https://github.com/llvm/llvm-project/blob/a6d366268d42469c77981e1c086d8035c9909990/clang/lib/Basic/Targets/AArch64.cpp#L487

I would have expected this macro to be defined here `clang/lib/Basic/Targets/SPIR.cpp` in `BaseSPIRVTargetInfo::getTargetDefines` right after:

https://github.com/llvm/llvm-project/blob/a6d366268d42469c77981e1c086d8035c9909990/clang/lib/Basic/Targets/SPIR.cpp#L61

Reply:
Thanks for the context! I'd defined it here because this was where I found the other HLSL macros, but I'll do it in `clang/lib/Basic/Targets/` to be consistent with other ≠how other targets are handled.